onnxruntime/python/onnxruntime_pybind_state.cc

```diff
@@ -227,7 +227,11 @@ const char* GetDeviceName(const OrtDevice& device) {
     case OrtDevice::CPU:
       return CPU;
     case OrtDevice::GPU:
+#ifdef USE_DML
+      return DML;
+#else
       return CUDA;
+#endif
     case OrtDevice::FPGA:
       return "FPGA";
     case OrtDevice::NPU:
```

Comment:
>NPU [](http://example.com/codeflow?start=20&length=3)

Do we need to consider this case for DML too?

Reply:
Good question. I'm not sure what this involves w.r.t NPUs and DML. I'm not a fan of these enums that we've been piggybacking on since the start because DML should be an abstraction on top of GPU/NPU, so it shouldn't really matter from the end-user whether the underlying hardware is a GPU or an NPU. They should be able to enum their adapters and specify which hardware index they want to run ORT on, regardless of whether it's a GPU or an NPU.

Reply:
I looked through the code, and if we support the NPU device eventually it should be through another PR. We use `GPU` when allocating memory and when creating the DML allocators, so it wouldn't work anyway if we enabled it here.